### PR TITLE
add cumulus upgrade doc

### PR DESCRIPTION
**Pull request by immarvin**
see http://10.3.5.21/xcat-doc/html/advanced/networks/onie_switches/os_cumulus/upgrade.html

### CODE REVIEW

File: docs/source/advanced/networks/onie_switches/os_cumulus/upgrade.rst [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+   xdsh switch1 'apt-get update && apt-get upgrade && reboot' 
+
+check the `/etc/os-release` file to make sure the Cumulus OS is upgrade ::
+
+   cat /etc/os-release
+
+
+2. Performing a binary (full image) install of the new version, using ONIE. If you expect to upgrade between major versions or if you have the binary image to upgrade to, this way is the recommended one. Binary install will erase all the configuration and data on the switch, so please pay special attention to backup your data and configuration files on the switch before you start upgrading.
+ 
+the step to perform a binary (full image) install of the new version is:
```
Comment (gurevichmark) [inline docs/source/advanced/networks/onie_switches/os_cumulus/upgrade.rst]: Should be:
"The steps to perform a binary (full image) install of the new version are:"

File: docs/source/advanced/networks/onie_switches/os_cumulus/upgrade.rst [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+ distribute the sources.list file to the switches to upgrade  with ``xdcp``, take "switch1" as an example here ::
+
+   xdcp switch1 /tmp/sources.list  /etc/apt/sources.list 
+
+ then invoke ``apt-get update`` and ``apt-get install`` on the switches to start package upgrade, a reboot might be needed after upgrading ::
+
+   xdsh switch1 'apt-get update && apt-get upgrade && reboot' 
+
+check the `/etc/os-release` file to make sure the Cumulus OS is upgrade ::
```
Comment (gurevichmark) [inline docs/source/advanced/networks/onie_switches/os_cumulus/upgrade.rst]: Instead of "is upgrade" better to say "has been upgraded"

File: docs/source/advanced/networks/onie_switches/os_cumulus/upgrade.rst [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+   cat /etc/os-release
+
+
+2. Performing a binary (full image) install of the new version, using ONIE. If you expect to upgrade between major versions or if you have the binary image to upgrade to, this way is the recommended one. Binary install will erase all the configuration and data on the switch, so please pay special attention to backup your data and configuration files on the switch before you start upgrading.
```
Comment (gurevichmark) [inline docs/source/advanced/networks/onie_switches/os_cumulus/upgrade.rst]: The last sentence is a little too wordy. Instead of 
"Binary install will erase all the configuration and data on the switch, so please pay special attention to backup your data and configuration files on the switch before you start upgrading."

Better to say:
"Make sure to backup your data and configuration files because binary install will erase all the configuration and data on the switch."

**Comment by cxhong** [comment]
License needs to be installed again.  otherwise,  the switchd will not able to restart.

**Comment by mattaezell** [comment]
Cumulus ZTP should put the license file back in place, but when I tried this earlier today it complained that it cannot find `cl-license`.  I am working on a MR that will fix that and several other problems I have found in my end-to-end test.

**Comment by immarvin** [comment]
hi @mattaezell , you might need to specify the full path of ``cl-license``

**Comment by immarvin** [comment]
hi @mattaezell , thanks for your comments, doc updated

**Comment by immarvin** [comment]
hi @gurevichmark , thanks for your comments, doc updated

**Comment by immarvin** [comment]
hi @cxhong , add the section to install license

**Comment by mattaezell** [comment]
#4005 fixes the license installation in ZTP.  It also adds a feature where ZTP can download the interface configuration file from the management node.

**Comment by immarvin** [comment]
hi @mattaezell , i will review your PR

**Comment by immarvin** [comment]
hi @cxhong @gurevichmark , if there is no other comment, please feel free to merge this, thanks
